core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1684,6 +1669,21 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /**
+   * Get the segments in the log which have a non-empty offset intersection with the
+   * inbound swap segment
+   */
+  private def swapSegmentIntersection(swapSegment: LogSegment): Iterable[LogSegment] = {
```

Comment:
I had made it generic initially, but it felt awkward since it doesn't make as much sense for an ordinary segment (the intersection of an ordinary segment is the segment itself). I thought it would be less confusing to tie it directly to swap handling.